src/app/core/navigation/navigation/navigation.component.ts

```diff
@@ -72,4 +76,30 @@ export class NavigationComponent {
     );
     return !viewConfig?.requiresAdmin || this.adminGuard.isAdmin();
   }
+
+  /**
+   * Listen for the installation prompt
+   */
+  @HostListener("window:beforeinstallprompt", ["$event"])
+  onbeforeinstallprompt(e) {
+    this.showInstallButton = true;
+    e.preventDefault();
+    this.deferredPrompt = e;
+  }
+
+  /**
+   * Show the installation prompt when button is clicked
+   */
+  public addToHomeScreen() {
+    this.deferredPrompt.prompt();
+    this.deferredPrompt.userChoice.then((choiceResult) => {
+      if (choiceResult.outcome === "accepted") {
+        console.log("User accepted the A2HS prompt");
+        this.showInstallButton = false;
+      } else {
+        console.log("User dismissed the A2HS prompt");
+      }
+      this.deferredPrompt = null;
```

Comment:
If the user denied the prompt the first time (`else` case), can the deferred prompt be called again to show the prompt a second time?

Reply:
Yes, the button still functions after the installation has been declined.